File: go/ex4_2_2.go
package main

type List[T any] struct { v T }
func (l List[T]) get(idx int) T { return l.v }

type Comparable[T any] interface {
    gt(that T) bool
}

func sort[T Comparable[T]](lst List[T]) {
    if (lst.get(0).gt(lst.get(1))) {}
}

type Person struct { age int }
func (this Person) gt(that Person) bool {
    return this.age > that.age
}

func main() {
    var people List[Person] = List[Person]{Person{0}}
    sort[Person](people)
}

File: java/ex4_2_2.java
import java.util.List;

abstract class Comparable<T> {
    abstract boolean gt(T that);
}

class A {
    static <T extends Comparable<T>> void sort(List<T> lst) {
        if (lst.get(0).gt(lst.get(1))) {}
    }
}

class Person extends Comparable<Person> {
    int age;
    boolean gt(Person that) { return this.age > that.age; }
}

class Example {
    public static void main(String[] args) {
        List<Person> people = List.of();
        A.<Person>sort(people);
    }
}
